src/vscode-bicep/schemas/bicepconfig.schema.json

```diff
@@ -0,0 +1,195 @@
+{
+    "$schema": "http://json-schema.org/schema#",
+    "title": "Bicep Configuration",
+    "definitions": {
+        "level": {
+            "title": "Diagnostic Level",
+            "description": "Type of diagnostic to display, defaults to warning",
+            "type": "string",
+            "default": "warning",
+            "enum": [
+                "off",
+                "info",
+                "warning",
+                "error"
+            ]
+        },
+        "rule": {
+            "type": "object",
+            "default": {
+                "level": "warning"
+            },
+            "properties": {
+                "level": {
+                    "$ref": "#/definitions/level"
+                }
+            },
+            "required": [
+                "level"
+            ]
+        }
+    },
+    "type": "object",
+    "default": {
+        "analyzers": {
+            "core": {
+                "enabled": true
+            }
+        }
+    },
+    "properties": {
+        "analyzers": {
+            "title": "Analyzers",
+            "type": "object",
+            "default": {
+                "core": {
+                    "enabled": true,
+                    "rules": {
+                        "no-hardcoded-env-urls": {
+                            "level": "warning"
+                        },
+                        "no-unused-params": {
+                            "level": "warning"
+                        },
+                        "no-unused-vars": {
+                            "level": "warning"
+                        },
+                        "prefer-interpolation": {
+                            "level": "warning"
+                        },
+                        "secure-parameter-default": {
+                            "level": "warning"
+                        },
+                        "simplify-interpolation": {
+                            "level": "warning"
+                        }
+                    }
+                }
+            },
+            "properties": {
+                "core": {
+                    "title": "Core Bicep Analyzer",
+                    "description": "Core analyzer built into bicep",
+                    "type": "object",
+                    "required": [
+                        "enabled"
+                    ],
+                    "properties": {
+                        "verbose": {
+                            "type": "boolean",
+                            "default": false
+                        },
+                        "enabled": {
+                            "type": "boolean",
+                            "default": true
+                        },
+                        "rules": {
```

Comment:
>rules [](http://example.com/codeflow?start=25&length=5)

We should have a test to fail the build if this file is not in sync with the rules that are implemented.

Reply:
good idea